spec/bundler_inject_spec.rb

```diff
@@ -0,0 +1,210 @@
+RSpec.describe Bundler::Inject do
+  shared_examples_for "overrides" do
+    it "with local overrides" do
+      write_bundler_d_file <<~F
+        gem "rack-obama"
+      F
+      bundle(:update)
+
+      expect(out).to match %r{^Injecting .+/bundler\.d/local_overrides\.rb\.\.\.$}
+      expect(lockfile_specs).to match_array [["rack", "2.0.6"], ["rack-obama", "0.1.1"]]
+    end
+
+    it "with global overrides" do
+      write_global_bundler_d_file <<~F
+        gem "rack-obama"
+      F
+      bundle(:update)
+
+      expect(out).to match %r{^Injecting .+/.bundler\.d/global_overrides\.rb\.\.\.$}
+      expect(lockfile_specs).to match_array [["rack", "2.0.6"], ["rack-obama", "0.1.1"]]
+    end
+
+    it "with local and global overrides" do
+      write_bundler_d_file <<~F
+        gem "rack-obama"
+      F
+      write_global_bundler_d_file <<~F
+        gem "omg"
+      F
+      bundle(:update)
+
+      expect(out).to match %r{^Injecting .+/bundler\.d/local_overrides\.rb\.\.\.$}
+      expect(out).to match %r{^Injecting .+/.bundler\.d/global_overrides\.rb\.\.\.$}
+      expect(lockfile_specs).to match_array [["rack", "2.0.6"], ["rack-obama", "0.1.1"], ["omg", "0.0.6"]]
+    end
+
+    describe "#override_gem" do
+      it "with a different version" do
+        write_bundler_d_file <<~F
+          override_gem "rack", "=2.0.5"
+        F
+        bundle(:update)
+
+        expect(lockfile_specs).to eq [["rack", "2.0.5"]]
+        expect(err).to match %r{^\*\* override_gem\("rack", "=2.0.5"\) at .+/bundler\.d/local_overrides.rb:1$}
+      end
+
+      it "with a git repo" do
+        write_bundler_d_file <<~F
+          override_gem "rack", :git => "https://github.com/rack/rack"
+        F
+        bundle(:update)
+
+        expect(lockfile_specs).to eq [["rack", extract_rack_version]]
+        expect(err).to match %r{^\*\* override_gem\("rack", :git=>"https://github.com/rack/rack"\) at .+/bundler\.d/local_overrides.rb:1$}
+      end
+
+      it "with a path" do
+        Dir.mktmpdir do |path|
+          path = Pathname.new(path)
+          Dir.chdir(path) { `git clone https://github.com/rack/rack 2>/dev/null` }
```

Comment:
Shell out to git clone from github?  Aren't there tiny git repos we can init with the data we need?

Reply:
maybe?  I need it to be a real gem, though.  rack is pretty small, actually

Reply:
I changed it to at least `git clone --depth 1`, which, for rack, is ~290Kb